File: waterball/ch1/practices/p1_dependency/py/main.py
from hero.hero import Hero
from hero.levelsheet import LevelSheet


def main():
    # Create a new Hero instance
    h = Hero()

    # Create a new LevelSheet instance
    ls = LevelSheet()

    # Test gaining experience
    exps = [0, 100, 900, -200]
    for exp in exps:
        try:
            h.gain_exp(exp, ls)
        except ValueError as e:
            print(e)


if __name__ == "__main__":
    main()

File: waterball/ch1/practices/p1_dependency/golang/main.go
package main

import (
	"fmt"
	"hero-dependency/hero"
)

func main() {
	myHero := hero.NewHero()
	levelSheet := hero.NewLevelSheet()

	// Test gaining experience
	exps := []int{0, 100, 900, -200}
	for _, exp := range exps {
		err := myHero.GainExp(exp, *levelSheet)
		if err != nil {
			fmt.Println(err)
			continue
		}
	}
}
